Gems/ROS2/Code/Source/FactorySimulation/ConveyorBeltComponentKinematic.h

```diff
@@ -0,0 +1,120 @@
+        //! @param handle the handle of the simulated body of the segment
+        //! @return the location of the segment in world space
+        AZ::Vector3 GetLocationOfSegment(AzPhysics::SceneHandle sceneHandle, AzPhysics::SimulatedBodyHandle handle);
+
+        //! Obtains the transform of of the pose on the spline at the given distance
+        //! @param splinePtr the spline to obtain the transform from
+        //! @param distanceNormalized the distance along the spline to obtain the transform from (normalized)
+        //! @return the transform of the pose on the spline at the given distance
+        AZ::Transform GetTransformFromSpline(AZ::ConstSplinePtr splinePtr, float distanceNormalized);
+
+        //! Spawn rigid body at the given location
+        //! @param splinePtr the spline to spawn the rigid body on
+        //! @param physicsSystem the physics system to spawn the rigid body in
+        //! @param sceneHandle the scene handle of the scene to spawn the rigid body in
+        //! @param location the location to spawn the rigid body at (normalized)
+        //! @return a pair of the normalized location and the handle of the simulated body
+        AZStd::pair<float, AzPhysics::SimulatedBodyHandle> CreateSegment(
+            AZ::ConstSplinePtr splinePtr,
+            AzPhysics::SystemInterface* physicsSystem,
+            AzPhysics::SceneHandle sceneHandle,
+            float normalizedLocation);
+
+        // AZ::TickBus::Handler overrides...
+        void OnTick(float delta, AZ::ScriptTimePoint timePoint) override;
+
+        //! Every physic update
+        void PostPhysicsSubTick(float fixedDeltaTime);
+        //! Every physic update - scene event handler
+        AzPhysics::SceneEvents::OnSceneSimulationFinishHandler m_sceneFinishSimHandler;
+        //! Speed of the conveyor belt
+        float m_speed = 1.0f;
+        //! Width of the conveyor belt
+        float m_beltWidth = 1.0f;
+        //! Length of individual segments of the conveyor belt
+        float m_segmentLength = 1.0f;
+        //! Material of individual segments of the conveyor belt
+        AZ::Data::Asset<Physics::MaterialAsset> m_materialAsset;
+        //! Cache of created segments
+        AZStd::deque<AZStd::pair<float, AzPhysics::SimulatedBodyHandle>> m_ConveyorSegments;
+        //! Heigh of belt
+        static constexpr float m_segmentWidth = 0.1f;
+        //! Offset of the texture
+        float m_textureOffset = 0.0f;
+        //! Scaling factor of the texture
+        float m_textureScale = 1.0f;
+        //! Pointer to the spline
+        AZ::ConstSplinePtr m_splineConsPtr{ nullptr };
+        //! Real spline length
+        float m_splineLength = -1.0f;
+        //! Transform from spline's local frame to world frame
+        AZ::Transform m_splineTransform;
+        //! Start and end point of the belt
+        AZ::Vector3 m_startPoint;
+        AZ::Vector3 m_endPoint;
```

Comment:
Please group members into clearly named structures of configuration.

Reply:
It is in logical order. please give me more details.